src/router/index.js

```diff
@@ -183,7 +183,11 @@ const routes = [
 
 const router = createRouter({
   history:
-    process.env.VUE_APP_ELECTRON_MODE === 'production'
+    /**
+     * For GitHub Pages deployment only.
+     * If variable is not available in process environment, use from the `.env` file instead.
+     */
+    (window.__VITE_ROUTER_HISTORY_MODE__ ?? import.meta.env.VITE_ROUTER_HISTORY_MODE === 'hash')
```

Comment:
Looks like overengineering. What's the purpose of `window.__VITE_ROUTER_HISTORY_MODE__`? Why not just using `import.meta.env.VITE_ROUTER_HISTORY_MODE`?

Reply:
`window.__VITE_ROUTER_HISTORY_MODE__` is from environment variable
`import.meta.env.VITE_ROUTER_HISTORY_MODE` is from the `.env` file
If `.env` file is not an option, it is not necessary

Reply:
Can you just use `import.meta.env.VITE_ROUTER_HISTORY_MODE` and remove `window.__VITE_ROUTER_HISTORY_MODE__` everywhere?

Reply:
Currently no because `VITE_ROUTER_HISTORY_MODE` is not in the `.env` file, it is [set dynamically](https://github.com/Adamant-im/adamant-im/blob/feature/github-pages-hash-routes/.github/workflows/github-pages.yml#L42) in workflow.
If return `echo 'VITE_ROUTER_HISTORY_MODE="hash"' >> .env` [back](https://github.com/Adamant-im/adamant-im/commit/054360b), `import.meta.env.VITE_ROUTER_HISTORY_MODE` will be sufficient

Reply:
https://github.com/Adamant-im/adamant-im/pull/778/commits/af83217b27d4849b08a8fa805cea564294da09a3

Reply:
Strange, it did not work in my profile deployments, so much the better